src/main/com/marginallyclever/makelangelo/plotter/plotterControls/SaveGCode.java

```diff
@@ -51,52 +46,48 @@ private String addExtension(String name, String [] extensions) {
 		return name + "." + extensions[0];
 	}
 	
-	private void save(OutputStream outputStream,Turtle turtle,Plotter robot) throws Exception {
+	private void save(String filename, Turtle turtle, Plotter robot) throws Exception {
 		Log.message("saving...");
 		
-		OutputStreamWriter out = new OutputStreamWriter(outputStream);
+		try (Writer out = new OutputStreamWriter(new FileOutputStream(filename))) {
```

Comment:
why is there a try here if there is no catch?  Why is it better to create the OutputStreamWriter here than where it was?

I love when format-all makes everything angry red....

Reply:
in the original implementation, the `close()` was not called if an exception was thrown is the method. Here, it is the [try with resources statement](https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html) which will close any stream open between the brace at the beginning.
I didn't add a catch because the method already throws Exception which tells that the parent is handling the exception for me.